src/main/java/team/chisel/inventory/InventoryChiselSelection.java

```diff
@@ -151,8 +167,11 @@ public ItemStack getStackInSlotOnClosing(int slot) {
 
 	@Override
 	public void setInventorySlotContents(int slot, ItemStack stack) {
-		inventory[slot] = stack;
-		updateInventoryState(slot);
+		// we do not want external chisel variant updates
```

Comment:
maybe there can be some drawbacks about this change
i will explain

chisel gui has item container both on server and client.
so i added scrolling packet to sync server's container.
otherwise server container is not updated, always on first page and chiseling result is wrong.

but when updating servers container it is synced back to client. so
```
client scroll 1
client draw 1
client scroll 2
client draw 2
server got scroll 1
server updates client container to 1
client draw 1
server got scroll 2
server updates client container to 1
client draw 2
```
so client scrolling is shaking and it is annoying

but i am not shure this is good way to fix this

Reply:
Add a flag to the server -> client packet that the sync reason is scrolling?

Reply:
it is vanila S2FPacketSetSlot and i cannot add flag without some dark magic
and what to do with this flag?

Reply:
I think rather than changing the container inventory, you could instead change the rendered/displayed slots range and having the container have all the items from the variation group. Then you don't have to bother syncing server and client. Inventory content is same both sides.

Reply:
good idea but may take more than one night to implement. i'll try